presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -346,13 +355,35 @@ private void addRolesRecords()
         }
     }
 
+    private void addRoleAuthorizationDescriptorRecords()
+    {
+        try {
+            accessControl.checkCanShowRoles(session.toSecurityContext(), catalogName);
+        }
+        catch (AccessDeniedException exception) {
+            return;
+        }
+
+        for (RoleGrant grant : metadata.listAllRoleGrants(session, catalogName, roles, grantees, limit)) {
+            addRecord(
+                    grant.getRoleName(),
+                    null, // grantor
```

Comment:
It would be nice to have these defined in `RoleGrant`, but being unused currently. So other connector implementations could benefit from that.

Reply:
Let's do that in a separate PR if that's ok.